services/server/src/handlers/topic.ts

```diff
@@ -170,6 +178,46 @@ async function addUserTopics(req: Request, res: Response, next: NextFunction) {
   res.sendStatus(200);
 }
 
+async function replaceUserTopics(req: Request, res: Response, next: NextFunction) {
+  const user = req.user;
+  const allowed = isStudent(user) || isTutor(user) || isTutorManager(user);
+  if (!allowed) return next(forbidden());
+
+  const { removeTopics, addTopics }: ITopic.ReplaceUserTopicsApiPayload = 
+    replaceUserTopicsBodyParser.parse(req.body);
+
+  if (isEmpty(removeTopics) || isEmpty(addTopics)) return next(empty());
+  // to not leave a vulnerability which can make the user exceed the max number
+  if (removeTopics.length !== addTopics.length) return next(bad()); 
+
+  // ensure that all oldTopics exists for the user
```

Comment:
You can verify that the user didn't exceeded the max allowed topic count by following this simple equation.

```
const allowed = (inDbCount + addTopics.legnth - removeTopics.legnth) <= MAX_ALLOWED_TOPICS 
```